internal/pkg/cli/run_local_test.go

```diff
@@ -411,198 +398,71 @@ func TestRunLocalOpts_Execute(t *testing.T) {
-				m.interpolator.EXPECT().Interpolate("").Return("", nil)
-
-				runCalled := make(chan struct{})
-				isRunningCalled := make(chan struct{})
-				m.dockerEngine.EXPECT().Run(gomock.Any(), expectedRunPauseArgs).DoAndReturn(func(ctx context.Context, opts *dockerengine.RunOptions) error {
-					close(runCalled)
-					<-isRunningCalled
-					return nil
-				})
-				m.dockerEngine.EXPECT().IsContainerRunning(gomock.Any(), mockPauseContainerName).DoAndReturn(func(ctx context.Context, name string) (bool, error) {
-					<-runCalled
-					defer close(isRunningCalled)
-					return false, errors.New("some error")
-				})
-
-				m.prog.EXPECT().Start(gomock.Any()).Return().Times(6)
-				m.prog.EXPECT().Stop(gomock.Any()).Return().Times(3)
-				m.dockerEngine.EXPECT().Stop(gomock.Any(), gomock.Any()).Return(nil).Times(3)
-				m.dockerEngine.EXPECT().Rm(gomock.Any()).Return(nil).Times(3)
-			},
-			wantedError: errors.New(`run pause container: check if container is running: some error`),
-		},
-		"error if fail to run service container": {
+		"pulls errors from orchestrator": {
 			inputAppName:  testAppName,
 			inputWkldName: testWkldName,
 			inputEnvName:  testEnvName,
-			setupMocks: func(m *runLocalExecuteMocks) {
+			setupMocks: func(t *testing.T, m *runLocalExecuteMocks) {
 				m.ecsLocalClient.EXPECT().TaskDefinition(testAppName, testEnvName, testWkldName).Return(taskDef, nil)
 				m.ssm.EXPECT().GetSecretValue(gomock.Any(), "mysecret").Return("secretvalue", nil)
 				m.ws.EXPECT().ReadWorkloadManifest(testWkldName).Return([]byte(""), nil)
 				m.interpolator.EXPECT().Interpolate("").Return("", nil)
 
-				runCalled := make(chan struct{})
-				m.dockerEngine.EXPECT().Run(gomock.Any(), expectedRunPauseArgs).DoAndReturn(func(ctx context.Context, opts *dockerengine.RunOptions) error {
-					close(runCalled)
-					return nil
-				})
-				m.dockerEngine.EXPECT().IsContainerRunning(gomock.Any(), mockPauseContainerName).DoAndReturn(func(ctx context.Context, name string) (bool, error) {
-					<-runCalled
-					return true, nil
-				})
-				m.dockerEngine.EXPECT().Run(gomock.Any(), expectedRunFooArgs).Return(errors.New("some error"))
-				m.dockerEngine.EXPECT().Run(gomock.Any(), expectedRunBarArgs).Return(nil)
-
-				m.prog.EXPECT().Start(gomock.Any()).Return().Times(6)
-				m.prog.EXPECT().Stop(gomock.Any()).Return().Times(3)
-				m.dockerEngine.EXPECT().Stop(gomock.Any(), gomock.Any()).Return(nil).Times(3)
-				m.dockerEngine.EXPECT().Rm(gomock.Any()).Return(nil).Times(3)
-			},
-			wantedError: errors.New(`run container "foo": some error`),
-		},
-		"success": {
+				errCh := make(chan error, 1)
+				m.orchestrator.StartFn = func() <-chan error {
+					errCh <- errors.New("some error")
+					return errCh
+				}
+				m.orchestrator.RunTaskFn = func(task orchestrator.Task) {
+					require.Equal(t, expectedTask, task)
+				}
+				m.orchestrator.StopFn = func() {
+					require.Len(t, errCh, 0)
```

Comment:
do you think it's necessary at all to test "how we are handling the error", which in our case, would mean "making sure we are really handling the error by printing it"?